bin/node/src/commands/net.rs

```diff
@@ -90,9 +93,9 @@ impl NetCommand {
                         Err(e) => debug!(target: "net", "Failed to receive unsafe payload: {:?}", e),
                     }
                 }
-                _ = interval.tick() => {
+                _ = interval.tick(), if tx.is_some() => {
                     let (otx, mut orx) = tokio::sync::oneshot::channel();
-                    if let Err(e) = tx.send(P2pRpcRequest::PeerCount(otx)).await {
+                    if let Err(e) = tx.as_ref().unwrap().send(P2pRpcRequest::PeerCount(otx)).await {
```

Comment:
Can we match this in the select to be `, Some(sender) = &tx => {` or does that not play nicely?

If not, let's at least move this up a few lines and avoid using `unwrap()`:

```rust
let Some(sender) = tx else {
    unreachable!("tx must be some");
};
```